spec/api-net-spec.js

```diff
@@ -364,6 +364,51 @@ describe('net module', function () {
       urlRequest.end()
     })
 
+    it('should be able to set a non-string object as header value', function (done) {
+      const requestUrl = '/requestUrl'
+      const customHeaderName = 'Some-Integer-Value'
+      const customHeaderValue = 900
+      server.on('request', function (request, response) {
+        switch (request.url) {
+          case requestUrl:
+            assert.equal(request.headers[customHeaderName.toLowerCase()],
+              customHeaderValue.toString())
+            response.statusCode = 200
+            response.statusMessage = 'OK'
+            response.end()
+            break
+          default:
+            assert(false)
```

Comment:
Maybe an `assert.equal(request.url, requestUrl)` before here would be better so when this fails we know what `request.url` was.